### PR TITLE
Refactor dialogs

### LINKED ISSUE #10: New "Configuration/settings" screen
Based on this mockup (see the "NOTE"s), we were envisioning the "mirror-selection" and "download location selection" settings could be combined in to one settings screen, and all settings there would be saved between QGIS sessions. The settings screen should only be automatically displayed once on first launch of the plugin, but accessible any time through a dedicated menu.

image

### LINKED ISSUE #19: Add messaging when download is complete
Maybe replace the 100% progress bar with a "Done! XXX MB downloaded" message? What do you think

### LINKED ISSUE #21: Progress bar shows 100% when re-entering the download screen
After downloading one set of data successfully, if I press "Prev" to go back to the data selection screen, select a second set of data, then click next to download the data, this screen shows 100%; instead it should indicate that the download has not started. Clicking "Download" does start the download as expected.

### LINKED ISSUE #22: Display chosen storage location in download screen
After using the plugin once, closing QGIS, re-opening it, and attempting to download more layers with the plugin, the download screen looks like this:

image

We would like this empty field to contain the chosen download location and for the download button to be immediately clickable -- in other words; the user should only need to select a download location once.

### PR DESCRIPTION
**Pull request by ghtmtt**
I tried to be as atomic as I could :)

Huge changes here!

* we have 2 different dialogs, one for the downloading and the other one for the server settings. If no server options have been set then a Warning dialog pops up and the server setting will be automatically opened
* some general file refactoring (file names and folders)
* geo rid of the `README.html` and `README.txt` files. Replaced with the `README.md` one
* some minor cleaning like the progress bar, total size of megabytes downloaded

Many issues are addressed. Of course some tests are needed!

Fix #10 fix #19 fix #21 fix #22

### CODE REVIEW

**Comment by MattF-NSIDC** [comment]
Matteo,

We haven't had a chance to look at the code yet, but we just got done playing with this in QGIS. Thank you for your excellent work. All of the linked issues are looking resolved on our end. Trey and I are feeling good about making this available to users as a beta now. We're going to have a chat with Twila and see if there are any tweaks she'd like before then.

We'll open a few more issues for future releases, and we'll reach out to you to let you know about the beta release. Thank you again, this is looking really good :grin:

Matt and Trey

**Review by MattF-NSIDC — APPROVED** [review]

File: qgreenland_server.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+
+
+class QGreenlandServer(QtWidgets.QDialog, FORM_CLASS):
+    def __init__(self, parent=None):
+        """Constructor."""
+        super(QGreenlandServer, self).__init__(parent)
+        self.setupUi(self)
+
+        self.settings = QgsSettings()
```
Comment (MattF-NSIDC) [inline qgreenland_server.py:46]: I'm still trying to learn how this works so thanks in advance for your help understanding :) Does `QgsSettings()` return the same settings object to `QGreenlandServer` as it does for `QGreenlandDownload`, or does it create a new instance?

File: qgreenland_server.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        # fill the combobox with the values
+        self.server_list_combo.addItem(self.tr('NSIDC: http://qgreenland.apps.nsidc.org/layers/'), 'http://qgreenland.apps.nsidc.org/layers/')
+        self.server_list_combo.addItem(self.tr('PGC: https://example.com/qgreenland'), 'https://example.com/qgreenland')
+
+        self.buttonBox.accepted.connect(self.write_settings)
```
Comment (MattF-NSIDC) [inline qgreenland_server.py:52]: This looks like it's registering a callback function that will be called when the accept button is clicked, right? Neat!

File: qgreenland_dowload.py [hunk 3]
```diff
@@ -607,7 +642,10 @@ def download_data(self):
                     f.write(reply_content)
             
             self.progressBar.setValue(int((current + 1) * total))
-
+        
+        # set the final progress bar text with the amount of megabytes downloaded
+        self.progressBar.setFormat(f"{total_size:,} MB have been downloaded")
```
Comment (MattF-NSIDC) [inline qgreenland_dowload.py:647]: What do you think about `{total_size:,.2f}` to limit the number of decimal places to 2?

File: qgreenland_dowload.py [hunk 4]
```diff
@@ -559,35 +598,31 @@ def download_data(self):
                 if layer['id'] == parent:
                     for asset in layer['assets']:
                         files_to_download.append(asset['file'])
+                        total_size+=asset['size_bytes']
+                        print(asset['size_bytes'])
                     layer_to_download[parent] = files_to_download
+        
+        # get the bytes in megabytes
+        total_size = total_size / 1000000
```
Comment (MattF-NSIDC) [inline qgreenland_dowload.py:606]: I recently learned about a trick to make large numbers more readable in python. You can break them up in to digit groups using underscores: `1_000_000` is the same as `1000000` :)

Comment (ghtmtt) [reply qgreenland_dowload.py:606]: oh thanks! I didn't know it!

File: qgreenland_dowload.py (hunk 3 above)
Comment (ghtmtt) [reply qgreenland_dowload.py:647]: yes, makes sense

File: qgreenland_server.py (hunk 1 above)
Comment (ghtmtt) [reply qgreenland_server.py:46]: basically with `QgsSettings` you are dealing with the settings of the profile folder of QGIS. So even if it is a new instance in another file (and class) you have access to the `QGIS.ini` file variables. Just have a look in the `QGIS` folder within the user profile and open the `QGIS.ini` file and search for QGreenland :)

File: qgreenland_server.py (hunk 2 above)
Comment (ghtmtt) [reply qgreenland_server.py:52]: yep exactly. `buttonBox` is the name of the *double* button (Cancel + OK) of Qt. You can connect to another function when the Cancel button is pressed (`self.buttonBox.rejected.connect(self.my_other_function`)

**Comment by ghtmtt** [comment]
632c30d  addresses the small *numeric* enhancements of the review. I'm going to merge the PR. Thanks for the review!

**Pull request merged**